astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -362,6 +364,100 @@ def test_astype(self):
             int32q = self.q.astype("int32")
             assert_array_equal(np.astype(int32q, "int32"), int32q)
 
+    @needs_array_function
+    @pytest.mark.parametrize(
+        "args, kwargs, expected",
+        [
+            pytest.param(
+                (1 * u.radian,),
+                {},
+                np.arange(1, dtype=float),
+                id="pos: stop",
+            ),
+            pytest.param(
+                (0 * u.degree, 1 * u.radian),
+                {},
+                np.arange(1, dtype=float),
+                id="pos: start, stop",
+            ),
+            pytest.param(
+                (0 * u.degree, 1 * u.radian, 1 * u.arcsec),
+                {},
+                np.arange(ARCSEC_PER_RADIAN, dtype=float),
+                id="pos: start, stop, step",
+            ),
+            pytest.param(
+                (0 * u.degree, 1 * u.radian),
+                {"step": 1 * u.arcsec},
+                np.arange(ARCSEC_PER_RADIAN, dtype=float),
+                id="pos: start, stop; kw: step",
+            ),
+            pytest.param(
+                (0 * u.radian,),
+                {"stop": 5 * u.radian},
+                np.rad2deg(np.arange(5, dtype=float) * ARCSEC_PER_DEGREE),
+                id="pos: start; kw: stop",
+            ),
+            pytest.param(
+                (10 * u.radian, None),
+                {},
+                np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
+                id="pos: stop, followed by 1 None",
+            ),
+            pytest.param(
+                (10 * u.radian, None, None),
+                {},
+                np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
+                id="pos: stop, followed by 2 None",
+            ),
+            pytest.param(
+                (10 * u.radian, None, None, None),
+                {},
+                np.rad2deg(np.arange(10, dtype=float) * ARCSEC_PER_DEGREE),
+                id="pos: stop, followed by 3 None",
+            ),
+        ],
+    )
+    def test_arange(self, args, kwargs, expected):
+        arr = np.arange(*args, **kwargs, like=u.Quantity([], u.rad))
+        assert type(arr) is u.Quantity
+        assert arr.unit == u.radian
+        assert arr.dtype == expected.dtype
+        np.testing.assert_array_almost_equal(arr.to_value(u.arcsec), expected)
```

Comment:
Might as well do an `from numpy.testing import assert_array_almost_equal, assert_array_equal` on top.

Reply:
ah, actually it's already there. Let me simplify this !